low-level/sensors/impl/generic/raid_integrity_data.py

```diff
@@ -166,22 +167,39 @@ def _raid_health_monitor(self):
                 # Check RAID device array state is 'idle' or not
                 result = self._check_raid_state(device)
                 if result == "failed":
+                    # Send alert
+                    self.alert_type = self.FAULT
+                    self._alert_msg = RaidAlertMsgs.STATE_MSG.value
+                    self._send_json_msg(self.alert_type, device, self._alert_msg)
+                    # Persist RAID device fault state
+                    self._fault_state = self.FAULT
+                    data = device + ":" + self._fault_state
+                    fault_status_file = self.DEFAULT_RAID_DATA_PATH + device + RaidDataConfig.RAID_SYNC_FAULT_STATUS.value
+                    with open(fault_status_file, 'w') as fs:
```

Comment:
same here, will keep writing same fault state to file, even if it didnt changed from last time

Reply:
It will store the FAULT state once in a file only.
Then the file will be updated only when state changes to FAULT_RESOLVED